lib/add-custom-resource.js

```diff
@@ -56,14 +54,15 @@ module.exports = function addCustomResource(domainName, basePath ,deploymentId)
       },
       BasePath: basePath,
       DomainName: domainName,
-      Stage: stage,
+      Stage: stageName,
       RestApi: {
         Ref: 'ApiGatewayRestApi'
       }
     },
     DependsOn: [
       'ApiGatewayRestApi',
       'CustomApiGatewayBasePathMappingFunction',
+      'ApiGatewayStage',
```

Comment:
This should only be added if an ApiGatewayStage was created, and even if so, you have to determine the logical ID of it. It won't necessarily be called ApiGatewayStage.